src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
@Luka967 Wrong again!

First off, my tactic is not solvable with two lines of code. Also since the maximum of cells is around 5000, according to the buffer size, it sets it the maximum amount of minions and other things. 


It is not good practice to have memory increase by alot when there are more clients. Also in reality, multiple binarywriters would be used for a player, so that probably means more bytes of memory are used then I estimated. If I was a server owner, I wouldnt use this then. And Multi-Ogar's main purpose is to be a server for the masses. 100 players > 3000 kilobytes of data


That was barbosik's fault. Allocation checks are supposed to be fast.


Multi-Ogar can do it too.


I dont use an environment. And Multi-Ogar's cpu graphs are horrible.



You can make BinaryWriter two times more efficient by fixing the toBinary code. (Basically, precalculate)